lib/routemap.c

```diff
@@ -1473,17 +1469,21 @@ enum rmap_compile_rets route_map_add_match(struct route_map_index *index,
 		route_map_notify_dependencies(index->map->name,
 					      RMAP_EVENT_CALL_ADDED);
 	}
+	if (type != RMAP_EVENT_MATCH_ADDED)
+		route_map_upd8_dependency(type, rule_key, index->map->name);
 
 	return RMAP_COMPILE_SUCCESS;
 }
 
 /* Delete specified route match rule. */
 enum rmap_compile_rets route_map_delete_match(struct route_map_index *index,
 					      const char *match_name,
-					      const char *match_arg)
+					      const char *match_arg,
+					      route_map_event_t type)
 {
 	struct route_map_rule *rule;
 	struct route_map_rule_cmd *cmd;
+	const char *rule_key;
```

Comment:
can we initialize this variable here as well?

Reply:
It's not required. I have removed it from route_map_add_match also.